cluster/centos/util.sh

```diff
@@ -316,10 +333,54 @@ function provision-node() {
 #   KUBE_TEMP
 function ensure-setup-dir() {
   kube-ssh "${1}" "mkdir -p ${KUBE_TEMP}; \
+                   mkdir -p ${KUBE_TEMP}/etcd-cert; \
                    sudo mkdir -p /opt/kubernetes/bin; \
                    sudo mkdir -p /opt/kubernetes/cfg"
 }
 
+function ensure-etcd-cert() {
```

Comment:
This seems a lot like https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L740

I wonder if @jszczepkowski would be interest in you moving that stuff up higher in the tree and making it reusable by more providers.